sys/crypto/c25519.c

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright (C) 2015 Jan Wagner <mail@jwagner.eu>
+ * Copyright (C) 2014 Daniel Beer <dlbeer@gmail.com>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     sys_crypto
+ * @{
+ *
+ * @file
+ * @brief       Implementation of the Curve25519 Diffie-Hellman function
+ *
+ * @author      Jan Wagner <mail@jwagner.eu>
+ * @author      Daniel Beer <dlbeer@gmail.com>
+ * @author      D. J. Bernstein
```

Comment:
Same question as with the ```@author NSA```: did you use parts of a reference implementation provided by Bernstein?